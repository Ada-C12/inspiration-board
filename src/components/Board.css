.board {
  display: flex;
  flex-wrap: wrap;
}

.validation-errors-display {
  text-align: center;
  margin-bottom: 2em;
}

.validation-errors-display__list {
  list-style-type: none;
  padding: 0;
}

.error-message {
  margin: 1rem 0;
  padding: 1rem;

  background-color: tomato;
  color: white;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}

.success-message {
  margin: 1rem 0;
  padding: 1rem;

  background-color: lightgreen;
  color: white;
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
}