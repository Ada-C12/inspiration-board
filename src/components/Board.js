import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';

import './Board.css';
import Card from './Card';
import NewCardForm from './NewCardForm';

class Board extends Component {
  constructor() {
    super();

    this.state = {
      boardName: "",
      cards: [],
    };
  }

  removeCard = (cardId) => {
    axios.delete(`${this.props.url}/cards/${cardId}`)
      .then((response) => {
        const cards = this.state.cards.filter((cardItem) => cardItem.card.id !== cardId);
        this.setState({ cards, });
      })
      .catch((error) => {
        this.setState({ error: error.message });
      });
  }

  addCard = (card) => {
    const url = `${this.props.url}/boards/${this.props.boardName}/cards`
    axios.post(url, card)
      .then((response) => {
        const updatedCards = this.state.cards;
        updatedCards.push(response.data);
        this.setState({
          cards: updatedCards,
          error: "",
        });
      })
      .catch((error) => {
        this.setState({
          error: error.message,
        });
      });
  }
  
  getUpdatedData = () => {
    const cardUrl = `${this.props.url}/boards/${this.props.boardName}/cards`;
    axios.get(cardUrl).then((response) => {
      this.setState({
        cards: response.data,
        boardName: this.props.boardName,
      })
    }).catch((error) => {
      this.setState({ error: 'Sorry, something went wrong' })
    })
  }

  componentDidMount() {
    this.getUpdatedData();
  }

  componentDidUpdate() {
    if (this.state.boardName !== this.props.boardName) {
      console.log("UPDATING...")
      this.getUpdatedData();
    }
  }

  render() {
    const cards = this.state.cards.map((cardItem, i) => {
      return (
        <section key={i}>
          <Card
            cardData={cardItem.card}
            removeCardCallBack={this.removeCard}>
          </Card>
        </section>)
    });

    return (
      <section>
        <h3>Board: {this.props.boardName}</h3>
        <NewCardForm addCardCallBack={this.addCard}></NewCardForm>

        <section className="board">{cards}</section>
      </section>
    )
  }

}

Board.propTypes = {
  url: PropTypes.string.isRequired,
  boardName: PropTypes.string.isRequired,
};
export default Board;
