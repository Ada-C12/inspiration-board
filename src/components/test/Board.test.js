import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Board from '../Board';

describe('Board', () => {

  test('matches existing snapshot', () => {
    //Arrange-Act

    const result = render(
      <Board baseURL="base" URL="abc" boardName="xyz"/>
    );

    //Assert
      expect(result.asFragment()).toMatchSnapshot();
      cleanup();
  })

});