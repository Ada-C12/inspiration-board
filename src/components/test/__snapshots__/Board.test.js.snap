// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board that it matches the existing snapshot 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Board
    </h1>
    <h1 />
    <form
      class="new-card-form"
    >
      <h3>
        Add a card
      </h3>
      <div>
        <label
          class="new-card-form--label"
          for="text"
        >
          Text: 
        </label>
        <input
          id="text"
          name="text"
          value=""
        />
      </div>
      <div>
        <label
          class="new-card-form--label"
          for="emoji"
        >
          Emoji: 
        </label>
        <select
          id="emoji"
          name="emoji"
        >
          <option
            value=""
          />
          <option
            value=""
          />
          <option
            value="heart_eyes"
          >
            😍
          </option>
          <option
            value="beer"
          >
            🍺
          </option>
          <option
            value="clap"
          >
            👏
          </option>
          <option
            value="sparkling_heart"
          >
            💖
          </option>
          <option
            value="heart_eyes_cat"
          >
            😻
          </option>
          <option
            value="dog"
          >
            🐶
          </option>
        </select>
      </div>
      <input
        class="btn btn-success new-card-form--submit"
        name="submit"
        type="submit"
        value="Add a Card"
      />
    </form>
  </div>
</DocumentFragment>
`;
